src/components/UNSTABLE__Pagination/__tests__/Pagination.spec.js

```diff
@@ -0,0 +1,414 @@
+    let pageSelectorValues = [
+      ...getByLabelText(/test page selector/i).options,
+    ].map(page => page.value);
+
+    // Expect the default, set by `pageSize` prop, to be 5:
+    expect(queryByDisplayValue('5').selected);
+
+    // Expect the default number of selectable pages to be 1-4,
+    // since there are 20 total items and currently 5 items per page:
+    expect(pageSelectorValues).toEqual(['1', '2', '3', '4']);
+
+    // Change the page sizer value with user action:
+    userEvent.selectOptions(pageSizer, '10');
+
+    // Expect the updated displayed page size to show `10`:
+    expect(queryByDisplayValue('10').selected);
+
+    // Update page selector values since total page size has changed:
+    pageSelectorValues = [...getByLabelText(/test page selector/i).options].map(
+      page => page.value
+    );
+
+    // Expect 1-2 to be the selectable pages,
+    // since there are now 10 items per page out of 20 total items.
+    expect(pageSelectorValues).toEqual(['1', '2']);
+  });
+
+  test('should update current page when page selector is changed', () => {
+    const { getByText, getByLabelText } = render(
+      <UNSTABLE__Pagination totalItems={10} pageSize={5} pageSizes={[5]}>
+        {({ currentPage, onSetPage, totalPages }) => (
+          <PageSelector
+            currentPage={currentPage}
+            labelText="test page selector"
+            onChange={event => onSetPage(event.target.value)}
+            totalPages={totalPages}
+          />
+        )}
+      </UNSTABLE__Pagination>
+    );
+
+    // Expect to see default text for page 1,
+    // showing 5 items per page out of 10 total items.
+    expect(getByText(/1–5 of 10 items/i)).toBeInTheDocument();
+
+    // Expect the default displayed current page to be page 1:
+    expect(getByLabelText(/test page selector/i).value).toEqual('1');
+
+    // Go to page 2:
+    userEvent.selectOptions(getByLabelText(/test page selector/i), ['2']);
+
+    // Expect the page selector to show that we are on page 2:
+    expect(getByLabelText(/test page selector/i).value).toEqual('2');
+
+    // Expect to see updated text showing current items on current page:
+    expect(getByText(/6–10 of 10 items/i)).toBeInTheDocument();
+  });
+
+  test('should go to the next page when "next" button is clicked', () => {
+    const { getByText, getByLabelText } = render(
+      <UNSTABLE__Pagination
+        forwardText="test next button"
+        totalItems={10}
+        pageSize={5}
+        pageSizes={[5]}
+      >
+        {({ currentPage, onSetPage, totalPages }) => (
+          <PageSelector
+            currentPage={currentPage}
+            labelText="test page selector"
+            onChange={event => onSetPage(event.target.value)}
+            totalPages={totalPages}
+          />
+        )}
+      </UNSTABLE__Pagination>
+    );
+
+    // Expect to see default text for page 1:
+    expect(getByText(/1–5 of 10 items/i)).toBeInTheDocument();
```

Comment:
Is it sufficient to omit this assertion, as it's covered in the previous test?

Reply:
This assertion is setting a baseline / default for the next series of user events and assertions.
I don't think it is necessary to valid the text, however. It is done many times throughout this testing suite so multiple tests would fail if that text suddenly changed. 👍